src/3d/materials/qgsphongtexturedmaterialsettings.cpp

```diff
@@ -202,9 +232,35 @@ void QgsPhongTexturedMaterialSettings::addParametersToEffect( Qt3DRender::QEffec
 
   Qt3DRender::QParameter *ambientParameter = new Qt3DRender::QParameter( QStringLiteral( "ambientColor" ), ambientColor );
   Qt3DRender::QParameter *specularParameter = new Qt3DRender::QParameter( QStringLiteral( "specularColor" ), mSpecular );
-  Qt3DRender::QParameter *shininessParameter = new Qt3DRender::QParameter( QStringLiteral( "shininess" ), mShininess );
+  Qt3DRender::QParameter *shininessParameter = new Qt3DRender::QParameter( QStringLiteral( "shininess" ), static_cast<float>( mShininess ) );
 
   effect->addParameter( ambientParameter );
   effect->addParameter( specularParameter );
   effect->addParameter( shininessParameter );
 }
+
+void QgsPhongTexturedMaterialSettings::setDiffuseTexture( const QVariant &diffuse )
+{
+  if ( mDiffuseTexture.value<Qt3DRender::QTexture2D *>() )
+  {
+    delete mDiffuseTexture.value<Qt3DRender::QTexture2D *>();
```

Comment:
This looks messy. It's deleted here, but not in the destructor? And what about when the settings are copied/cloned? There's also no mention of lifetime or ownership in the docs. 

Could you add a smart pointer to hold the actual object instead of relying on the raw pointer stored in a variant?

Reply:
I was completely wrong. When the `diffuse` component is set from a `Qt3DExtras::QTextureMaterial`, the underlying `QImage` needs to be copied because the material takes ownership of the associated parameter. It worked before the QTexture2D from the settings was not deleted..

Now: 
- the `QImage` is correctly copied and used to create a texture parameter used by the material in `toMaterial`
- The `QTexture2D` of the settings is stored in a shared pointer.